### PR TITLE
Generate client code for new CRD AzureBackendPool

### PR DESCRIPTION
**Pull request by 3quanfeng**
1. Added CRD AzureBackendPool yaml definition files as well as examples
2. Added api types and client code generated by code-generator for the new CRD
3. Add one method, GetBackendPool in context.go

### CODE REVIEW

File: crds/AzureBackendPool.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: azurebackendpools.appgw.ingress.azure.io
+spec:
+  group: appgw.ingress.azure.io
+  versions:
+    - name: v1
```
Comment (akshaysngupta) [inline crds/AzureBackendPool.yaml:8]: v1 -> v1alpha1 ?

File: crds/AzureBackendPool.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                        description: "Subresource ID of the backend pool of the Application gateway"
+                        type: string
+                      ipAddresses:
+                        description: "A list of Pod IP addresses"
+                        type: array
+                        items:
+                          type: string
+  scope: Cluster
+  names:
+    plural: azurebackendpools
```
Comment (akshaysngupta) [inline crds/AzureBackendPool.yaml:34]: should this be application gateway backend pool ? (so as not to confuse with SLB's backend pool)

File: crds/AzureBackendPool.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: azurebackendpools.appgw.ingress.azure.io
+spec:
+  group: appgw.ingress.azure.io
```
Comment (akshaysngupta) [inline crds/AzureBackendPool.yaml:6]: Just to bring to notice, That we cannot use `azureingressprohibitedtargets.appgw.ingress.k8s.io` which was used for ProhibitedTarget CRD as sufficx `k8s.io` has been locked down.
https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190612-crd-group-protection.md

File: crds/AzureBackendPool.yaml (hunk 2 above)
Comment (3quanfeng) [reply crds/AzureBackendPool.yaml:34]: > should this be applicationgatewaybackendpool ? (so as not to confuse with SLB's backend pool)
Better name :)

File: crds/AzureBackendPool.yaml (hunk 1 above)
Comment (3quanfeng) [reply crds/AzureBackendPool.yaml:8]: > v1 -> v1alpha1 ?
Sounds good.
Here is some link I find regarding the versioning
https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/

Comment (akshaysngupta) [reply crds/AzureBackendPool.yaml:8]: will check this out

File: crds/AzureBackendPool.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  versions:
+    - name: v1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
```
Comment (jaishals) [inline crds/AzureBackendPool.yaml:15]: We need to add aproperty for the resource Id of application gateway ?

File: crds/AzureBackendPool.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+              type: object
+              properties:
+                backendPoolInfo:
```
Comment (jaishals) [inline crds/AzureBackendPool.yaml:18]: We need to add updateId  or GenerationId to correlate what update we are talking about?

File: crds/AzureBackendPool.yaml (hunk 4 above)
Comment (3quanfeng) [reply crds/AzureBackendPool.yaml:15]: Good point

File: pkg/k8scontext/context.go [hunk 6]
```diff
@@ -199,6 +200,26 @@ func (c *Context) GetAGICPod(envVariables environment.EnvVariables) *v1.Pod {
...
+	azpool, err := c.crdClient.AzureapplicationgatewaybackendpoolsV1beta1().AzureApplicationGatewayBackendPools().Get(backendPoolName, metav1.GetOptions{})
+	if err != nil {
+		glog.Error("Error fetching Azure application gateway backend pool resource, Error: ", err)
+		return nil
+	}
+	return azpool
+}
+
+// GetProhibitedTarget returns prohibited target with specified name and namespace
+func (c *Context) GetProhibitedTarget(namespace string, targetName string) *prohibitedv1.AzureIngressProhibitedTarget {
```
Comment (akshaysngupta) [inline pkg/k8scontext/context.go:214]: Let's remove these functions in a later PR.

**Pull request merged**